src/components/ChattyLLM/ChattyLLMInputForm.vue

```diff
@@ -219,20 +219,69 @@ export default {
 
 	watch: {
 		async active() {
+			this.loading.llmGeneration = false
+			this.loading.titleGeneration = false
 			this.allMessagesLoaded = false
 			this.chatContent = ''
 			this.msgCursor = 0
 			this.messages = []
 			this.editingTitle = false
 			this.$refs.inputComponent.focus()
 
-			if (this.active != null && !this.loading.newSession) {
+			if (this.active !== null && !this.loading.newSession) {
 				await this.fetchMessages()
 				this.scrollToBottom()
 			} else {
 				// when no active session or creating a new session
 				this.allMessagesLoaded = true
 				this.loading.newSession = false
+				return
+			}
+			// start polling in case a message is currently being generated
+			try {
+				const sessionId = this.active.id
+				const checkSessionResponse = await axios.get(getChatURL('/check_session'), { params: { sessionId } })
+				if (checkSessionResponse.data?.sessionTitle && checkSessionResponse.data?.sessionTitle !== this.active.title) {
+					this.active.title = checkSessionResponse.data?.sessionTitle
+					console.debug('update session title with check result')
+				}
+				console.debug('check session response:', checkSessionResponse.data)
+				if (checkSessionResponse.data.messageTaskId !== null) {
+					try {
+						this.loading.llmGeneration = true
+						const message = await this.pollGenerationTask(checkSessionResponse.data.messageTaskId, sessionId)
+						console.debug('checkTaskPolling result:', message)
+						this.messages.push(message)
+						this.scrollToBottom()
+					} catch (error) {
+						console.error('checkGenerationTask error:', error)
+						showError(t('assistant', 'Error generating a response'))
+					} finally {
+						this.loading.llmGeneration = false
+					}
+				} else if (checkSessionResponse.data.titleTaskId !== null) {
```

Comment:
`if` would be good here since there might be message and title tasks both scheduled